lbrynet/daemon/Daemon.py

```diff
@@ -3082,29 +3073,245 @@ def jsonrpc_stream_availability(self, uri, search_timeout=None, blob_timeout=Non
+                        )
+                    args = {
+                        'confirmations': confirmations,
+                        'include_reserved': include_reserved
+                    }
+                    if include_claims:
+                        args['include_claims'] = True
+                    return account.get_balance(**args)
+            raise Exception("Couldn't find an account named: '{}'.".format(account_name))
+        else:
+            if include_claims:
+                raise Exception("'--include-claims' requires specifying an LBC account.")
+            return self.wallet.get_balances(confirmations)
+
+    @requires("wallet")
+    def jsonrpc_max_address_gap(self, account_name):
+        """
+        Finds ranges of consecutive addresses that are unused and returns the length
+        of the longest such range: for change and receiving address chains. This is
+        useful to figure out ideal values to set for 'receiving_gap' and 'change_gap'
+        account settings.
+
+        Usage:
+            max_address_gap (<account_name> | --account=<account_name>)
+
+        Options:
+            --account=<account_name>        : (str) account for which to get max gaps
+
+        Returns:
+            (map) maximum gap for change and receiving addresses
+        """
+        return self.get_account_or_error('account', account_name).get_max_gap()
+
+    @requires("wallet")
+    def jsonrpc_fund(self, to_account, from_account, amount=0,
+                         everything=False, outputs=1, broadcast=False):
+        """
+        Transfer some amount (or --everything) to an account from another
+        account (can be the same account). Amounts are interpreted as LBC.
+        You can also spread the transfer across a number of --outputs (cannot
+        be used together with --everything).
+
+        Usage:
+            fund (<to_account> | --to_account=<to_account>)
+                 (<from_account> | --from_account=<from_account>)
+                 (<amount> | --amount=<amount> | --everything)
+                 [<outputs> | --outputs=<outputs>]
+                 [--broadcast]
+
+        Options:
+            --to_account=<to_account>     : (str) send to this account
+            --from_account=<from_account> : (str) spend from this account
+            --amount=<amount>             : (str) the amount to transfer lbc
+            --everything                  : (bool) transfer everything (excluding claims), default: false.
+            --outputs=<outputs>           : (int) split payment across many outputs, default: 1.
+            --broadcast                   : (bool) actually broadcast the transaction, default: false.
+
+        Returns:
+            (map) maximum gap for change and receiving addresses
+
+        """
+        to_account = self.get_account_or_error('to_account', to_account)
+        from_account = self.get_account_or_error('from_account', from_account)
```

Comment:
Hrm, where are you storing the mapping from named accounts to their index?  Just noting that those will also be lost in the case of seed recovery.

Reply:
You may be thinking of accounts under a single HD chain. Here each "Account" is a root of an HD chain. Basically, you can have more than one HD root in the same wallet.

Reply:
Right.  BTW, which BIP32 derivation are you following?

Reply:
The new wallet is backwards compatible with the old one in terms of how the `receiving` and `change` address chains are defined. Which is what we inherited from electrum. I don't believe it follows the BIP32 spec exactly.

The code for this is here:
https://github.com/lbryio/torba/blob/master/torba/bip32.py
https://github.com/lbryio/torba/blob/master/torba/baseaccount.py

Reply:
Electrum is definitely BIP32 compatible.